woocommerce/woocommerce
# Change csv export saveAs implementation library

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).

### Changes proposed in this Pull Request:

- Change csv export package's `saveAs` implementation from browser-filesaver to file-saver as browser-filesaver is not maintained anymore.
- Remove typings as they are not needed anymore, add `@types/file-saver`

This PR was originally opened in 2022 and has now been rebased onto current `trunk`.



### How to test the changes in this Pull Request:

1. Check out this branch and run `pnpm install`.
2. Run the package's unit tests
   ```sh
   cd packages/js/csv-export
   pnpm test:js
   ```
3. Confirm types still resolve without the removed `typings/index.d.ts`:
   ```sh
   pnpm lint:lang:types
   ```
4. Verify the download works end to end in a browser: go to **WooCommerce > Analytics > Orders**, apply filters so there is only a single page of data, then click **Download**. The CSV is generated client-side through the changed code path and should download as before.



### Testing that has already taken place:

Rebased onto `trunk` and verified locally (Node 24, pnpm 10.33.0):

-   `pnpm test:js` in `packages/js/csv-export`: passed
-   `pnpm lint:lang:types`: passes
-   `pnpm build`: passes
-   `pnpm lint:lang:js`: 0 errors. Two warnings remain (`no-undef` at `src/index.ts:140` and `FileReader` at `src/test/index.ts:121`), both on lines untouched by this PR
-   Lockfile regenerated with pnpm 10.33.0

### Milestone






### Changelog entry



The changelog entry is already committed in this branch.

### Release Communication



Neither option applies: this is an internal dependency swap with no user-facing behaviour change and no change to the package's public API.
